container/go/cmd/digester/digester.go

```diff
@@ -88,32 +93,92 @@ func main() {
 		}
 	}
 
-	img, err := utils.ReadImage(imgSrc, *format)
+	img, err := utils.ReadImage(imgSrc, *format, *tarball, layers)
 	if err != nil {
 		log.Fatalf("Error reading from %s: %v", imgSrc, err)
 	}
 
-	digest, err := img.Digest()
-	if err != nil {
-		log.Fatalf("Error getting image digest: %v", err)
+	configPath := ""
+	if *format == "legacy"{
+		configPath = *src
 	}
-
-	err = writeDigest(digest, *dst)
+	err = writeDigest(img, *dst, configPath)
 	if err != nil {
 		log.Fatalf("Error outputting digest file to %s: %v", *dst, err)
 	}
 
 	log.Printf("Successfully generated image digest file at %s", *dst)
 }
 
-// writeDigest outputs digest to a "digest file" at dst. Digest is the image manifest's sha256 hash.
-func writeDigest(digest v1.Hash, dst string) error {
+// writeDigest outputs image's digest to a "digest file" at dst. Digest is the image manifest's sha256 hash.
```

Comment:
Can be simplified to `writeDigest writes the sha256 digest of the manifest of the given image to the file given by dst. The image config is replaced with the config at the given path before generating the digest.`